cpp/box/Box.h

```diff
@@ -337,12 +346,15 @@ class Box
      *  \param Nvecs Number of vectors
      *  \param out The array in which to place the wrapped vectors.
      */
-    void wrap(const vec3<float>* vecs, unsigned int Nvecs, vec3<float>* out) const
+    void wrapPython(nb_array<float, nb::shape<-1, 3>> vecs, unsigned int Nvecs,
```

Comment:
```suggestion
    void wrapPython(nb_array<float, nb::shape<-1, 3>> vecs,
```
This (and similar) methods should discover `N` from the shape of the array. The duplicated information could be inconsistent, leading to out of bounds array accesses or bugs where only part of an array is read.

Reply:
Also: When the output must be the same size as the input, this (and similar methods) must verify that is the case.